packages/modules/src/modules/epics/ContributionsEpic.tsx

```diff
@@ -253,6 +261,72 @@ const EpicBody: React.FC<BodyProps> = ({
     );
 };
 
+/*
+Issues I'm having:
+
+    - This code properly belongs as a function in server/src/lib/ab.ts 
+    but I've struggled too long with types to get it working there
+
+    - Messaging test variants can suppress the Choice Cards. Need to check that 
+    the new code accounts for this and doesn't record users as being part of
+    an amounts test if they see a contribs epic with no choice card displayed?
+
+    - This code reflects (as far as possible) what happens in support frontend - 
+    as amounts testing can also start on the contributions page there if 
+    a user navigates to the page by means other than the Epic CTA button
+
+    - Can I generate the mvtId number locally here? Currently it's generated 
+    by the support frontend site and stored in a cookie there. I can't find 
+    where SDC generates the equivalent, though it seems to live in the 
+    tracking object after it has been generated
+
+    - How to make sure that if a user gets assigned to amounts testing, 
+    that testing persists for subsequent views of contributions epics? 
+    (I assume that this is a testing requirement)
+
+    - How do we track when a user sees an amounts test? Do we need to track
+    that a user has been assigned to an amounts test variant before they see
+    the Epic messaging? Or only after they reach that point?
+*/
+const MVT_MAX = 1000000;
+function randomNumber(mvtId: number, seed: number): number {
```

Comment:
you can use the existing logic in ab.ts instead, it uses the same library (seedrandom) - https://github.com/guardian/support-dotcom-components/blob/main/packages/server/src/lib/ab.ts#L17